colearn_pytorch/pytorch_learner.py

```diff
@@ -81,6 +100,24 @@ def __init__(self, model: torch.nn.Module,
         self.minimise_criterion = minimise_criterion
         self.vote_criterion = vote_criterion
 
+        self.dp_config = diff_priv_config
+        self.dp_privacy_engine = PrivacyEngine()
+
+        if diff_priv_config is not None:
+            (
+                self.model,
+                self.optimizer,
+                self.train_loader,
+            ) = self.dp_privacy_engine.make_private_with_epsilon(
+                module=self.model,
+                optimizer=self.optimizer,
+                data_loader=self.train_loader,
+                epochs=self.num_train_batches,
```

Comment:
Epochs and batches are not to be mixed.
Check if we really need to use the `make_private_with_epsilon` function. I think it is better to use the `make_private`, as the learner has no clue, how many training rounds will happen.
- as I understand (but can be wrong), the _with_epsilon calculates the noise_multiplier if you know your privacy budget and the number of epochs
- while the make_private lets you set the noise_multiplier, and with come accountant you can calculate the spend privacy based on the batch size/number of iterations

Reply:
updated the privacy engine to use `make_private`